src/tree.rs

```diff
@@ -90,12 +87,50 @@ where
     }
 
     /// Get the root of the merkle tree.
-    ///
-    /// It is none if the tree is empty.
     pub fn root(&self) -> Ref<T> {
         self.root.item(0)
     }
 
+    /// Returns the root of the smallest sub-tree that holds all the leaves.
+    ///
+    /// Returns None when the tree is empty. In this case use
+    /// `T::EMPTY_SUBTREES[H - 1]` for the root instead.
```

Comment:
Any reason to not return `T::EMPTY_SUBTREES[H - 1]` if the tree is empty? 
This way, `root()` and `smallest_subtree` would be consistent

Reply:
I would love to but couldn't find a way. The `root()` returns a `Ref<T>`, the `T` that is stored in the smallest node that holds all leaves. But when the tree is empty, that node doesn't exist so we can not return a `Ref` to it.

The reason why we need `Ref<T>` as the item type of each node, is the lazy computation that @ureeves implemented in #37 .

If you have a trick on how to make `smallest_subtree()` return only a `Ref<T>` (without the `Option`) or even only `&T`, I would love to hear it. I myself aren't completely happy with the current implementation.

Reply:
Since the tree is empty, the root of the tree already return `T::EMPTY_SUBTREES[n]`

So

```rust
            if children == 0 {
                return (self.root(), 0);
            }
```